opm/common/OpmLog/Logger.cpp

```diff
@@ -38,6 +38,18 @@ namespace Opm {
         addMessageType( Log::MessageType::Bug , "bug");
     }
 
+    void Logger::addTaggedMessage(int64_t messageType, const std::string& tag, const std::string& message) const {
+        if ((m_enabledTypes & messageType) == 0)
+            throw std::invalid_argument("Tried to issue message with unrecognized message ID");
+
+        if (m_globalMask & messageType) {
+            for (auto iter = m_backends.begin(); iter != m_backends.end(); ++iter) {
```

Comment:
Yes it could be a range for. I just copied addMessage() and did not change its implementation.

And yes, the no-tag version could just forward. As done now that forwarding gets done only at the lowest accessible level (the LogBackend), changing it here means duplicating that tiny piece of login with a small risk of inconsistency, but it's not a big deal. Will change it.